knowledge_graphs/query_pagination.py

```
@@ -0,0 +1,382 @@
+        """
+        # Remove skip and limit from params for count query
+        count_params = {k: v for k, v in params.items() if k not in ['skip', 'limit']}
+        
+        # Convert query to count query
+        count_query = self._convert_to_count_query(query)
+        
+        try:
+            result = await self.session.run(count_query, count_params)
+            count_record = await result.single()
+            return count_record['count'] if count_record else 0
+        except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)